internal/controllers/extension_controller.go

```diff
@@ -162,21 +164,29 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 	ext.Status.ResolvedBundle = bundleMetadataFor(bundle)
 	setResolvedStatusConditionSuccess(&ext.Status.Conditions, fmt.Sprintf("resolved to %q", bundle.Image), ext.GetGeneration())
 
+	// Populate the deprecation status using the resolved bundle
+	SetDeprecationStatusInExtension(ext, bundle)
+
 	mediaType, err := bundle.MediaType()
 	if err != nil {
-		setInstalledStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
-		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
+		if c := apimeta.FindStatusCondition(ext.Status.Conditions, ocv1alpha1.TypeInstalled); c == nil {
+			ext.Status.InstalledBundle = nil
+			setInstalledStatusConditionFailed(&ext.Status.Conditions, fmt.Sprintf("failed to read bundle mediaType: %v", err), ext.GetGeneration())
+		}
+		setProgressingStatusConditionFailed(&ext.Status.Conditions, fmt.Sprintf("failed to read bundle mediaType: %v", err), ext.GetGeneration())
```

Comment:
A little confusion here: `setProgressingStatusConditionFailed` is setting Progressing too true. As discussed [here](https://github.com/operator-framework/operator-controller/pull/683#discussion_r1540999472), if there is an error and we are not able to proceed further, then though an error, we are in a stable state with no other attempt.

Reply:
That makes sense, updated the PR so that we're only ever progressing if we're potentially unstable.